crates/cast/src/cmd/call.rs

```diff
@@ -26,9 +29,34 @@ use foundry_evm::{
     opts::EvmOpts,
     traces::{InternalTraceMode, TraceMode},
 };
-use std::str::FromStr;
+use regex::Regex;
+use std::{str::FromStr, sync::LazyLock};
+
+// matches override pattern <address>:<slot>:<value>
+// e.g. 0x123:0x1:0x1234
+static OVERRIDE_PATTERN: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"^([^:]+):([^:]+):([^:]+)$").unwrap());
 
 /// CLI arguments for `cast call`.
+///
+/// ## State Override Flags
+///
+/// The following flags can be used to override the state for the call:
+///
+/// * `--override-balance <address>:<balance>` - Override the balance of an account
+/// * `--override-nonce <address>:<nonce>` - Override the nonce of an account
+/// * `--override-code <address>:<code>` - Override the code of an account
+/// * `--override-state <address>:<slot>:<value>` - Override a storage slot of an account
+///
+/// Multiple overrides can be specified for the same account. For example:
+///
+/// ```bash
+/// cast call 0x... "transfer(address,uint256)" 0x... 100 \
+///   --override-balance 0x123:0x1234 \
+///   --override-nonce 0x123:1 \
+///   --override-code 0x123:0x1234 \
+///   --override-state 0x123:0x1:0x1234
```

Comment:
nit: missing --override-state-diff

Reply:
added in https://github.com/foundry-rs/foundry/pull/10255/commits/6bce07d8700eaeba3ebb5a22bd28b6d8c8c30960